src/main/treeDataProviders/utils/analyzerUtils.ts

```diff
@@ -186,57 +198,150 @@ export class AnalyzerUtils {
+     * @param fileScanBundles - the file bundles that contains all the information on scan results
      * @param scanProgress - the progress for the given scan
      */
     public static async cveApplicableScanning(
         scanManager: ScanManager,
-        fileScanBundle: FileScanBundle,
-        scanProgress: GraphScanProgress
+        fileScanBundles: FileScanBundle[],
+        progressManager: StepProgress
     ): Promise<void> {
-        let cvesToScan: Set<string> = new Set<string>();
-        if (!(fileScanBundle.dataNode instanceof ProjectDependencyTreeNode)) {
+        let spaceToBundles: Map<string, Map<FileScanBundle, Set<string>>> = this.mapBundlesForApplicableScanning(
+            scanManager.logManager,
+            fileScanBundles
+        );
+        if (spaceToBundles.size == 0) {
             return;
         }
-        let descriptorIssues: DependencyScanResults = <DependencyScanResults>fileScanBundle.data;
-        fileScanBundle.dataNode.issues.forEach((issue: IssueTreeNode) => {
-            if (issue instanceof CveTreeNode && !issue.parent.indirect && issue.cve?.cve) {
-                cvesToScan.add(issue.cve.cve);
+        for (let [spacePath, bundles] of spaceToBundles) {
+            let cveToScan: Set<string> = Utils.combineSets(Array.from(bundles.values()));
+            // Scan workspace for all cve in relevant bundles
+            let startApplicableTime: number = Date.now();
+            let applicableIssues: ApplicabilityScanResponse = await scanManager.scanApplicability(spacePath, progressManager.checkCancel, cveToScan);
+            if (applicableIssues && applicableIssues.applicableCve) {
+                let applicableScanTimestamp: number = Date.now();
+                AnalyzerUtils.transferApplicableResponseToBundles(
+                    applicableIssues,
+                    bundles,
+                    scanManager.logManager,
+                    applicableScanTimestamp,
+                    applicableScanTimestamp - startApplicableTime
+                );
             }
-        });
-        if (cvesToScan.size == 0) {
-            return;
         }
-        scanManager.logManager.logMessage('Scanning descriptor ' + descriptorIssues.fullPath + ' for cve applicability issues', 'INFO');
+    }
 
-        let startApplicableTime: number = Date.now();
-        descriptorIssues.applicableIssues = await scanManager.scanApplicability(
-            path.dirname(descriptorIssues.fullPath),
-            scanProgress.onProgress,
-            cvesToScan
-        );
+    /**
+     * Create a mapping between a workspace and all the given bundles that relevant to it.
+     * In addition, filter not relevant bundles.
+     * @param logManager - logger to log added map
+     * @param fileScanBundles - bundles to map
+     * @returns mapped bundles to similar workspace
+     */
+    private static mapBundlesForApplicableScanning(
```

Comment:
This function's name clearly implies its purpose: to perform mappings and nothing else. Other tasks like filtering out indirect elements are not relevant in this context. We should separate them into a new function to maintain clarity.